docs/scratch-deploy.md

```diff
@@ -0,0 +1,189 @@
+- Deploy Lido custom Aragon apps implementations (aka bases), namely for `Lido`, `LegacyOracle`, `NodeOperatorsRegistry`)
+- Registry Lido APM name in ENS
+- Deploy Aragon package manager contract `APMRegistry` (via `LidoTemplate`)
+- Deploy Lido custom Aragon apps repo contracts (via `LidoTemplate`)
+- Deploy Lido DAO (via `LidoTemplate`)
+- Issue DAO tokens (via `LidoTemplate`)
+- Deploy non-Aragon Lido contracts: `OracleDaemonConfig`, `LidoLocator`, `OracleReportSanityChecker`, `EIP712StETH`, `WstETH`, `WithdrawalQueueERC721`, `WithdrawalVault`, `LidoExecutionLayerRewardsVault`, `StakingRouter`, `DepositSecurityModule`, `AccountingOracle`, `HashConsensus` for AccountingOracle, `ValidatorsExitBusOracle`, `HashConsensus` for ValidatorsExitBusOracle, `Burner`.
+- Finalize Lido DAO deployment: issue unvested LDO tokens, setup Aragon permissions, register Lido DAO name in Aragon ID (via `LidoTemplate`)
+- Initialize non-Aragon Lido contracts
+- Set parameters of `OracleDaemonConfig`
+- Setup non-Aragon permissions
+- Plug NodeOperatorsRegistry as Curated staking module
+- Transfer all admin roles from deployer to `Agent`
+  - OZ admin roles: `Burner`, `HashConsensus` for `AccountingOracle`, `HashConsensus` for `ValidatorsExitBusOracle`, `StakingRouter`, `AccountingOracle`, `ValidatorsExitBusOracle`, `WithdrawalQueueERC721`, `OracleDaemonConfig`
+  - OssifiableProxy admins: : `LidoLocator`, `StakingRouter`, `AccountingOracle`, `ValidatorsExitBusOracle`, `WithdrawalQueueERC721`
+  - `DepositSecurityModule` owner
+
+## Local deployment
+
+Deploys the DAO to local (http://127.0.0.1:8545) dev node (anvil, hardhat, ganache).
+The deployment is done from default test account `0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266`.
+The node must be configured with the default test accounts derived from mnemonic `test test test test test test test test test test test junk`.
+
+1. Run `yarn install` (get sure repo dependencies are installed)
+2. Run the node on default port 8545 (for the commands see subsections below)
+3. Set test account private key `0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80` to `accounts.json` under `/eth/local` like `"local": ["<private key>"]` (see `accounts.sample.json` for example)
+4. Run the deploy script `bash scripts/scratch/dao-local-deploy.sh` from root repo directory
+5. Check out the deploy artifacts in `deployed-local.json`
+
+### Anvil
+
+Run the node with command:
+
+```shell
+anvil -p 8545 --auto-impersonate --gas-price 0 --base-fee 0 --chain-id 1337 --mnemonic "test test test test test test test test test test test junk"
+```
+
+### Hardhat node
+
+> NB: Hardhat node configuration is set in `hardhat.config.js` under `hardhat: { `.
+
+To run hardhat node execute:
+```shell
+yarn hardhat node
+```
+
+## Holešky deployment
+
+To do Holešky deployment, the following parameters must be set up via env variables:
+
+- `DEPLOYER`. The deployer address, you must have its private key. It must have enough ether.
```

Comment:
how much is enough tho?

Reply:
Approx 100,000,000 gas. Added the info to scratch-deploy.md